### ISSUE TITLE
Cocoapods plugin does not create DataStore category

### ISSUE THREAD

**Issue by sebsto** [aws-amplify/amplify-ios#244]
When using this schema

```graphql
type LandmarkObject
    @model
    @auth(rules: [ {allow: private, provider: userPools, operations: [ read ] } ])
{
    id: ID!
    name: String!
    category: String
    city: String
    state: String
    isFeatured: Boolean
    isFavorite: Boolean
    park: String
    coordinatesLat: Float
    coordinatesLong: Float
    imageName: String
}
```

and this data in the DDB table (created by an AppSync mutation) 

```json
{
  "__typename": "LandmarkObject",
  "category": "Lakes",
  "city": "Errol",
  "coordinatesLat": 44.747408,
  "coordinatesLong": -71.056816,
  "createdAt": "2019-12-09T18:58:53.381Z",
  "id": "9999",
  "imageName": "umbagog",
  "isFavorite": false,
  "isFeatured": true,
  "name": "Lake Umbagog",
  "park": "Umbagog National Wildlife Refuge",
  "state": "New Hampshire",
  "updatedAt": "2019-12-09T18:58:53.381Z"
}
```

and this swift code 

```swift 
    func queryLandmarks() {
        print("query landmarks...")
        Amplify.DataStore.query(LandmarkObject.self) {
            switch $0 {
                case .success(let result):
                    print("Landmarks: \(result)")   //result will be of type [Post]
            
                case .failure(let err):
                    print("Error listing LandmarkObjects - \(err.localizedDescription)")
            }
        }
    }
```

DatStore returns an empty array

```text
query landmarks...
Landmarks: []
```

I also found weird that only the model table is created in DDB, not the DatStore versioning table.

**Comment by sebsto** [comment]
Looks like the DataStore category is not created correctly 

My Pod file is like this 
```text 
target 'Landmarks' do
  # Comment the next line if you don't want to use dynamic frameworks
  use_frameworks!

  # Pods for Landmarks
  pod 'amplify-tools'
  pod 'Amplify'
  pod 'AWSPluginsCore'
  pod 'AmplifyPlugins/AWSAPIPlugin'
  pod 'AmplifyPlugins/AWSDataStorePlugin'

  pod 'AWSMobileClient', '~> 2.12.0'      # Required dependency
  pod 'AWSAuthUI', '~> 2.12.0'            # Optional dependency required to use drop-in UI
  pod 'AWSUserPoolsSignIn', '~> 2.12.0' 
```

But when building the project, the DataStore category is not created in amplify 

```sh
$ ls -al amplify/backend/
total 16
drwxr-xr-x  7 stormacq  ANT\Domain Users   224  9 Dec 19:07 .
drwxr-xr-x  8 stormacq  ANT\Domain Users   256  9 Dec 19:04 ..
-rw-r--r--@ 1 stormacq  ANT\Domain Users  2727  9 Dec 19:07 amplify-meta.json
drwxr-xr-x  3 stormacq  ANT\Domain Users    96  9 Dec 16:28 api
drwxr-xr-x  3 stormacq  ANT\Domain Users    96  9 Dec 15:25 auth
drwxr-xr-x  3 stormacq  ANT\Domain Users    96  9 Dec 15:27 awscloudformation
-rw-r--r--  1 stormacq  ANT\Domain Users   501  9 Dec 16:29 backend-config.json
```

**Comment by rodrigoelp** [comment]
I can confirm this is not working as intended (even in a plain model).

My sample project has been configured with the api and datastore plugin, querying the api returns some data I have provided prior the test, querying via the datastore I get nothing.

```swift
import Foundation
import Combine
import Amplify

let disposeBag = Set<AnyCancellable>()

func init() {
  // assume all this is part of the AppDelegate and you have all the files properly configured
        let models = AmplifyModels()
        let apiPlugin = AWSAPIPlugin(modelRegistration: models)
        let dataStorePlugin = AWSDataStorePlugin(modelRegistration: models)
        do {
            try Amplify.add(plugin: apiPlugin)
            try Amplify.add(plugin: dataStorePlugin)
            try Amplify.configure()
        } catch {
            print("Error initializing AppSync client. \(error)")
        }
}

func query() {
        dataStore.query(Post.self,
                        completion: { event in
                            switch event {
                                case .success(let items):
                                    print("DataStore items...")
                                    print(items)  // ==> These are always []
                                default:
                                    print("Oops?")
                            }
        })
        api.queryAsync(from: Post.self, where: nil)
            .sink(receiveCompletion: { _ in return }) { items in
                print("Plain API items...")
                print(items)      // ===> These are returned.
        }.store(in: &disposeBag)
}

query()
```

**Comment by sebsto** [comment]
@rodrigoelp : looks like it works when strictly following steps in the doc https://aws-amplify.github.io/docs/ios/datastore#using-cocoapods : 

1/ Podfile + pod install 
2/ XCode build
3/ Add model 
4/ Change `amplifyxc.config` => `modelgen = true` (see https://github.com/aws-amplify/amplify-ios/issues/255) 
5/ XCode build again
6/ add code for init and CRUD operation 
7/ optional push to the cloud (either via `amplifyxc.config` + xcode build or `amplify push`

**Comment by sebsto** [comment]
@rodrigoelp I received confirmation from product team that DataStore sync with the API is not fully functional in iOS yet.

**Comment by drochetti** [comment]
@rodrigoelp @sebsto This should be fixed in [Amplify 1.0.0](https://github.com/aws-amplify/amplify-ios/releases/tag/v1.0.0).

Feel free to re-open if you're still seeing this issue in the stable release.
